mlflow/store/cli.py

```diff
@@ -16,19 +27,19 @@ def commands():
 @commands.command("log-artifact")
 @click.option("--local-file", "-l", required=True,
               help="Local path to artifact to log")
-@click.option("--run-id", "-r", required=True,
```

Comment:
The CLI is now changed to require one of `--run-id` and `--base-artifact-uri`. This mimics the idea that a user may have either, as exposed from the MlflowClient (except in the case of the Java API, it always uses the `--base-artifact-uri` CLI, so we could remove the run-id variant -- but it is surprisingly difficult to use that variant by hand today).